pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+import signal
+import sys
+from signal import Signals
+from subprocess import call
+from types import FrameType, FunctionType
+from typing import Callable, List, Union
+
+import pytorch_lightning as pl
+from pytorch_lightning.utilities.imports import _fault_tolerant_training
+
+log = logging.getLogger(__name__)
+
+
+class HandlersCompose:
+    def __init__(self, signal_handlers: Union[List[Callable], Callable]):
+        if not isinstance(signal_handlers, list):
+            signal_handlers = [signal_handlers]
+        self.signal_handlers = signal_handlers
+
+    def __call__(self, signum: Signals, frame: FrameType) -> None:
+        for signal_handler in self.signal_handlers:
+            signal_handler(signum, frame)
+
+
+class SignalConnector:
+    def __init__(self, trainer: "pl.Trainer"):
+        self.trainer = trainer
+        self.trainer._terminate_gracefully = False
+
+    def register_signal_handlers(self) -> None:
+        sigusr1_handlers: List[Callable] = []
+        sigterm_handlers: List[Callable] = []
+
+        if _fault_tolerant_training():
+            sigusr1_handlers.append(self.fault_tolerant_sigusr1_handler_fn)
```

Comment:
why do we need them only for sigurs1? shouldn't they apply to all signals?

Reply:
This is the current registered signal. We can explore why one we should support, but in another PR.